lib/lifecycles/bump.js

```diff
@@ -120,7 +131,34 @@ function bumpVersion (releaseAs, currentVersion, args) {
         debug: args.verbose && console.info.bind(console, 'conventional-recommended-bump'),
         preset: presetOptions,
         path: args.path,
-        tagPrefix: args.tagPrefix
+        tagPrefix: args.tagPrefix,
+        whatBump: (commits) => {
+          let level = PATCH
+          let breakings = 0
+          let features = 0
+
+          commits.forEach(commit => {
```

Comment:
I think this is most of the way there, but one concern, the conventional-commits strategy has a more complex `whatBump` implementation:

https://github.com/conventional-changelog/conventional-changelog/blob/8076d4666c2a3ea728b95bf1e4e78d4c7189b1dc/packages/conventional-changelog-conventionalcommits/conventional-recommended-bump.js

Which includes support for `!` in the commit body.

Is it possible for us to use whatever default `whatBump` is configured already, and then update the `reason` afterwards?

Reply:
Initially, I had that approach in mind but after some exploring, I was not able to find a clear way to get the default `whatBump` implementation, ending up implementing my custom `whatBump` :/ 

💭  Would be cool if `conventional-recommended-bump` passed the default `whatBump` callback  as an option argument  to the `whatBump` option, `whatBump(commits, { defaultWhatBump, ...}) {};` 💭 
[It send a bunch of options](https://github.com/conventional-changelog/conventional-changelog/blob/8076d4666c2a3ea728b95bf1e4e78d4c7189b1dc/packages/conventional-recommended-bump/index.js#L86) but none we can use.

How do you think we should proceed @bcoe ?
I can adapt the current implementation to follow the strategy you linked, to support the `!` use case you mentioned.

Reply:
@nelsonfncosta I think each preset should be exporting its `recommendedBump` configuration, see:

https://github.com/conventional-changelog/conventional-changelog/blob/8076d4666c2a3ea728b95bf1e4e78d4c7189b1dc/packages/conventional-changelog-conventionalcommits/index.js#L29

The problem with copying the logic over from any one preset, is that we allow folks to configure their own preset -- so although the logic would be good for conventional commits, it would be off potentially for another plugin.

I haven't tried to do this myself, and apologize for how much of a yarn ball the conventional changelog codebase is -- I think though you should be able to pluck `whatBump` off of a given preset.

Reply:
I will take a closer look into this! 👀 
The information we need is on the `recommendedBumpOpts`, so if this is being exported it should be a straightforward change 👍

Reply:
I have been looking for a way to get the preset config
something along the lines

```
const options = {} // mostly the info in args 
const config = require('conventional-changelog-conventionalcommits')

const { recommendedBumpOpts } = config(options)
```

the problem I'm facing I'm not really sure how to "wait/get" my value from [here](https://github.com/conventional-changelog/conventional-changelog/blob/8076d4666c2a3ea728b95bf1e4e78d4c7189b1dc/packages/conventional-changelog-conventionalcommits/index.js#L36)